nomos-services/mempool/src/backend/mockpool/mod.rs

```diff
@@ -146,9 +146,10 @@ where
     }
 
     fn view(&self, _ancestor_hint: BlockId) -> Box<dyn Iterator<Item = Self::Item> + Send> {
-        // we need to have an owned version of the iterator to bypass adding a lifetime
-        // bound to the return iterator type
-        #[expect(clippy::needless_collect)]
+        #[expect(
+            clippy::needless_collect,
+            reason = "We need to have an owned version of the iterator to bypass adding a lifetime bound to the return iterator type"
+        )]
```

Comment:
This actually could be solved now with the `use` capture keyword.

Reply:
This is a bigger change because it seems to imply a change in the function signature of the trait, which would have cascading effects in other parts of the code as well, so I will not tackle it here.